content/tutorials/build-a-data-source-backend-plugin.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Build a backend for a data source plugin
+summary: Create a backend for your data source plugin.
+id: build-a-data-source-backend-plugin
+categories: ["plugins"]
+tags: beginner
+status: Published
+authors: Grafana Labs
+Feedback Link: https://github.com/grafana/tutorials/issues/new
+weight: 70
+---
+
+{{< tutorials/step duration="1" title="Introduction" >}}
+
+Grafana supports a wide range of data sources, including Prometheus, MySQL, and even Datadog. There's a good chance you can already visualize metrics from the systems you have set up. In some cases, though, you already have an in-house metrics solution that you’d like to add to your Grafana dashboards. This tutorial teaches you to build a support for your data source.
+
+In this tutorial, you'll:
+
+- Build a backend for your data source
+- Implement a health check for your data source
+- Enable alerting for your data source
+
+### Prerequisites
+
+- Knowledge about how data sources are implemented in the frontend.
+- Grafana version 7.0+
+- Go 1.14+
+- [Mage](https://magefile.org/)
+
+{{< /tutorials/step >}}
+{{< tutorials/step duration="1" title="Setup Your Environment" >}}
+
+{{< tutorials/shared "set-up-environment" >}}
+
+{{< /tutorials/step >}}
+{{< tutorials/step duration="1" title="Create a new plugin" >}}
+
+To build a backend for your data source plugin, Grafana requires a binary that it can execute when it loads the plugin during start-up. In this guide, we will build a binary using our backend plugin SDK in Go.
+
+The easiest way to get started is to clone one of our test data datasources. Navigate to the plugin folder that you configured in step 1 and type:
+
+```
+npx @grafana/toolkit plugin:create my-plugin
+```
+
+select `Backend Datasource Plugin` and follow the rest of the steps in the plugin scaffolding command.
+
+```
+cd my-plugin
+```
+
+To build the plugin:
+
+```
+yarn install --pure-lockfile
+ yarn build
+ mage -v buildAll
+```
+
+{{< /tutorials/step >}}
+{{< tutorials/step duration="1" title="Anatomy of a Backend Plugin" >}}
```

Comment:
[We should use sentence case for our titles](https://github.com/grafana/tutorials/pull/39/files#r424298675):
```suggestion
{{< tutorials/step duration="1" title="Anatomy of a backend plugin" >}}
```